apps/web/src/pages/templates/components/ListProviders.tsx

```diff
@@ -11,165 +13,142 @@ import { stepNames } from '../constants';
+  return '/static/images/providers/' + colorScheme + '/square/' + providerId + '.svg';
+};
+
+const ListProvidersContainer = styled.div`
+  margin-bottom: 2rem;
+  overflow: hidden;
+`;
+
+const TitleAndButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  color: ${colors.B60};
+  margin-bottom: 0.75rem;
+  font-size: 0.875rem;
+  line-height: 1.25rem;
+`;
+
+const ConfigureButton = styled(Button)`
+  height: 2rem;
+  padding: 0.5rem 1rem;
+`;
+
+const IntegrationButton = styled(UnstyledButton)<
+  { isActive: boolean; isDark: boolean } & { onClick?: MouseEventHandler<HTMLButtonElement> }
+>`
+  width: 100%;
+  padding: 0.5rem 0.75rem;
+  background: ${({ isDark }) => (isDark ? colors.B20 : colors.B98)};
+  border-radius: 0.5rem;
+  margin-bottom: 0.75rem;
+  line-height: 1;
+  opacity: ${({ isActive, isDark }) => (isActive || isDark ? 1 : 0.4)};
+`;
+
+const IntegrationIcon = styled.img<{ isActive: boolean; isDark: boolean }>`
+  height: 1.5rem;
+  max-width: 8.75rem;
+  opacity: ${({ isActive, isDark }) => (isActive || isDark ? 1 : 0.4)};
+`;
+
+const IntegrationStatusStyled = styled(IntegrationStatus)`
+  min-width: 5rem;
+`;
+
 export const ListProviders = ({
   channel,
-  providers,
+  channelProviders,
   setConfigureChannel,
   setProvider,
 }: {
   channel: ChannelTypeEnum;
-  providers: IIntegratedProvider[];
+  channelProviders: IIntegratedProvider[];
   setConfigureChannel: (channel: ChannelTypeEnum) => void;
   setProvider: (provider: IIntegratedProvider) => void;
 }) => {
   const { colorScheme } = useMantineColorScheme();
   const { environment: currentEnvironment } = useEnvController();
+  const containsNovuProvider = useMemo(
+    () =>
+      NOVU_SMS_EMAIL_PROVIDERS.some(
+        (providerId) => providerId === channelProviders.find((provider) => provider.connected)?.providerId
+      ),
+    [channelProviders]
+  );
+  const providersForTheCurrentEnvironment = useMemo(
+    () =>
+      channelProviders.filter((provider) => provider.connected && provider.environmentId === currentEnvironment?._id),
+    [channelProviders, currentEnvironment?._id]
+  );
+  const hasProviders = providersForTheCurrentEnvironment.length > 0;
+  const hasNovuProvider = providersForTheCurrentEnvironment.length === 1 && containsNovuProvider;
+  const isDark = colorScheme === 'dark';
 
   return (
-    <div
-      style={{
-        marginBottom: 32,
-        overflow: 'hidden',
-      }}
-    >
-      <div
-        style={{
-          color: colors.B60,
-          marginBottom: 12,
-          fontSize: 14,
-          lineHeight: '20px',
-        }}
-      >
-        <Group position="apart">
-          <ChannelTitle spacing={8} channel={channel} />
-          <Button
-            sx={{
-              height: '32px',
-              padding: '7.5px 15px',
-            }}
-            variant={providers.filter((provider) => provider.connected).length > 0 ? 'outline' : 'gradient'}
+    <ListProvidersContainer>
```

Comment:
I could spend probably more time refactoring it, but I don't think it's worth it and to me, the current state is better than what we had before.